The past few years have seen significant advances in the study of complex microbial communities associated with the evolution of sequencing technologies and increasing adoption of whole genome shotgun sequencing methods over the once more traditional Amplicon-based methods.
Through this evolution, these approaches have been built with associated computational strategies to tackle the data they generate. However,
these computational methods have not been generally accompanied by thoughtful design strategies that prioritise long term support with low maintainability, high data accessibility and end-to-end automation. 

In this work, we aim to first elaborate on the computational landscape in metagenomics, and how its methods can disregard fundamental software development principles that guide them towards improved reproducibility, principles such as dependency isolation, high parameterization, automatic report generation, with interactive figures that facilitate data exploration and, finally, descriptive and practical documentation. Following this, we tackle current limitations in metagenomic data processing by implementing a new data analysis pipeline, EURYALE, based on a previous methodology, MEDUSA, that selected its tools through rigorous benchmarking. This new pipeline, adaptable to different scenarios and built with good software development practices as guiding principles, serves to advance metagenomic data processing as a whole, and, additionally, make the data resulting from these analysis pipelines accessible to a wider audience.